package.json

```diff
@@ -64,63 +72,111 @@
-    "@babel/plugin-proposal-class-properties": "^7.1.0",
-    "@babel/plugin-transform-runtime": "^7.1.0",
-    "@babel/preset-env": "^7.1.0",
-    "@babel/preset-react": "^7.0.0",
-    "@babel/register": "^7.0.0",
+    "@babel/core": "^7.4.3",
     "@reactions/component": "^2.0.2",
-    "@storybook/addon-actions": "^3.4.11",
-    "@storybook/addon-links": "^3.4.11",
-    "@storybook/addon-options": "^3.4.11",
-    "@storybook/addons": "^3.4.11",
-    "@storybook/react": "^3.4.11",
+    "@storybook/addon-actions": "^4.1.16",
+    "@storybook/addon-links": "^4.1.16",
+    "@storybook/addon-options": "^4.1.16",
+    "@storybook/addons": "^4.1.16",
+    "@storybook/react": "^4.1.16",
+    "@types/enzyme": "^3.9.1",
+    "@types/enzyme-adapter-react-16": "^1.0.5",
+    "@types/react-test-renderer": "^16.8.1",
+    "@types/starwars-names": "^1.6.0",
+    "@types/storybook__react": "^4.0.1",
+    "@typescript-eslint/eslint-plugin": "^1.6.0",
+    "@typescript-eslint/parser": "^1.6.0",
     "ava": "^1.3.1",
-    "babel-core": "7.0.0-bridge.0",
-    "babel-eslint": "^10.0.1",
-    "babel-plugin-add-react-displayname": "^0.0.5",
+    "awesome-typescript-loader": "^5.2.1",
+    "babel-core": "^6.26.0",
+    "babel-loader": "^8.0.5",
     "concurrently": "^4.0.1",
     "enzyme": "^3.7.0",
     "enzyme-adapter-react-16": "^1.6.0",
-    "eslint-config-prettier": "^3.1.0",
-    "eslint-config-xo-react": "^0.17.0",
-    "eslint-plugin-react": "^7.11.1",
+    "eslint-config-prettier": "^4.1.0",
+    "eslint-config-xo-react": "^0.19.0",
+    "eslint-config-xo-typescript": "^0.9.0",
+    "eslint-plugin-react": "^7.12.4",
+    "eslint-plugin-react-hooks": "^1.6.0",
     "execa": "^1.0.0",
     "faker": "^4.1.0",
     "file-loader": "^2.0.0",
     "fs-extra": "^7.0.0",
-    "husky": "^1.1.2",
-    "lint-staged": "^7.3.0",
     "np": "^4.0.2",
-    "prettier": "^1.14.3",
-    "react": "^16.5.2",
+    "prettier": "^1.16.4",
+    "react": "^16.8.6",
     "react-chromatic": "^0.8.4",
-    "react-dom": "^16.5.2",
-    "react-test-renderer": "^16.5.2",
+    "react-docgen-typescript-loader": "^3.1.0",
+    "react-docgen-typescript-webpack-plugin": "^1.1.0",
+    "react-dom": "^16.8.6",
+    "react-test-renderer": "^16.8.6",
     "sinon": "^7.0.0",
     "size-limit": "^0.21.1",
     "starwars-names": "^1.6.0",
+    "storybook-addon-jsx": "^5.3.0",
+    "ts-node": "^8.0.3",
+    "typescript": "^3.4.1",
     "url-loader": "^1.1.2",
     "xo": "^0.24.0"
   },
   "xo": {
-    "parser": "babel-eslint",
+    "parser": "@typescript-eslint/parser",
+    "typescript": true,
+    "ignores": [
+      "docs",
+      "tools",
+      "src/**/stories",
+      "src/**/test",
+      "**/*.js",
+      "src/**/*.d.ts"
+    ],
     "extends": [
+      "xo-typescript",
       "xo-react",
       "prettier",
       "prettier/react"
     ],
+    "extensions": [
+      "tsx",
+      "ts"
+    ],
     "envs": [
       "browser"
     ],
-    "space": true,
+    "4space": true,
```

Comment:
Is this change intentional? What is `4space`?

Reply:
oops! not intentional